### PR TITLE
Move spec/support to lib/haml_lint to publish

### PR DESCRIPTION
**Pull request by LukeeeeBennett**
With these changes, users can require useful spec/support modules to use for custom linters in their own repo.

```
require 'haml_lint/spec/normalize_indent'
require 'haml_lint/spec/shared_linter_context'
require 'haml_lint/spec/matchers/report_lint'
```

or

```
require 'haml_lint/spec'
```

Closes https://github.com/sds/haml-lint/issues/302

### CODE REVIEW

File: .rubocop.yml [hunk 1]
```diff
@@ -65,6 +67,12 @@ Metrics/AbcSize:
 Metrics/BlockLength:
   Exclude:
     - 'spec/**/*.rb'
+    - 'lib/haml_lint/spec/**/*.rb'
+
+Metrics/ModuleLength:
```
Comment (LukeeeeBennett) [inline .rubocop.yml]: Unfortunately a couple lines over because of a couple whitespace changes to make line length work.

**Comment by LukeeeeBennett** [comment]
@sds This is ready for review. I made the move in a separate commit, so if you view the HEAD commit without whitespace changes it will show you every change I made to each file to get this working.

I'll note that I also wasn't sure about whether to register each component with rspec at require time in its own file (which I ended up doing) or just export a module and let importers set it up. My ruby fu is still too limited to see how to do this with the shared context or matcher so have just left it. 😂  

Ahh, maybe I dont have to worry about the context and matcher because they depend on rspec, but the normalize indent helper is a generic module and should be `include`d manually downstream?

**Comment by LukeeeeBennett** [comment]
Also coveralls fails but coverage is the same. 😢 I'm guessing we can ignore that.

File: spec/spec_helper.rb [hunk 2]
```diff
@@ -15,6 +15,9 @@
 Rainbow.enabled = false
 
 require 'haml_lint'
+require 'haml_lint/spec/normalize_indent'
+require 'haml_lint/spec/shared_linter_context'
+require 'haml_lint/spec/matchers/report_lint'
```
Comment (sds) [inline spec/spec_helper.rb]: You could create `lib/haml_lint/spec.rb` to contain these lines and then you would only need:

```ruby
require 'haml_lint/spec'
```

Might be helpful for other consumers as well.

I don't know how folks will use this, but I think it's a safe assumption that if you're `require`ing this that you want to have HAML Lint spec helpers injected into your RSpec runtime.

Comment (LukeeeeBennett) [reply spec/spec_helper.rb]: Awesome thanks :D

**Comment by LukeeeeBennett** [comment]
@sds Back to you :)

**Comment by LukeeeeBennett** [comment]
Thank you for being super responsive with guidance, review and release! 🎉 🏆

**Pull request merged**